R/importSNV.TCGA.R

```diff
@@ -0,0 +1,323 @@
+                               na.strings = " ")[,7]
+            strand_ <- read.table(i, header = TRUE, fill = NA, stringsAsFactors = FALSE,
+                                  na.strings = " ")[,8]
+            var_class_ <- read.table(i, header = TRUE, fill = NA, stringsAsFactors = FALSE,
+                                     na.strings = " ")[,9]
+            var_type_ <- read.table(i, header = TRUE, fill = NA, stringsAsFactors = FALSE,
+                                    na.strings = " ")[,10]
+            ref_allele_ <- read.table(i, header = TRUE, fill = NA, stringsAsFactors = FALSE,
+                                      na.strings = " ")[,11]
+            tum_all1_ <- read.table(i, header = TRUE, fill = NA, stringsAsFactors = FALSE,
+                                    na.strings = " ")[,12]
+            tum_all2_ <- read.table(i, header = TRUE, fill = NA,  stringsAsFactors = FALSE,
+                                    na.strings = " ")[,13]
+            tum_samp_barcode_ <- read.table(i, header = TRUE, fill = NA,  stringsAsFactors = FALSE,
+                                            na.strings = " ")[,16]
+            tum_sam_uuid_ <- read.table(i, header = TRUE, fill = NA,  stringsAsFactors = FALSE,
+                                        na.strings = " ")[,33]
+
+            # check if variant_class contains valid values from known list
+            classVknown <- c("Frame_Shift_Del", "Frame_Shift_Ins", "In_Frame_Del", "In_Frame_Ins", "Missense_Mutation",
+                        "Nonsense_Mutation", "Silent", "Splice_Site", "Translation_Start_Site", "Nonstop_Mutation",
+                        "3'UTR", "3'Flank", "5'UTR", "5'Flank", "IGR1" , "Intron", "RNA", "Targeted_Region")
+            myVclasses <- unique(var_class_)
+            if (! all(myVclasses %in% classVknown)) {
+                # drop file
+                classValid <- FALSE
+            } else {
+                classValid <- TRUE
+            }
+
+            # check if vairant_type contains valid values from known list
+            typeVknown <- c("SNP", "DNP", "TNP", "ONP", "INS", "DEL", "Consolidated2")
+            myVtypes <- unique(var_type_)
+            if (! all(myVtypes %in% typeVknown)) {
+                # drop file
+                typeValid <- FALSE
+            } else {
+                typeValid <- TRUE
+            }
+
+            # check chromosome value is valid
+            typeChr <- c("chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13",
+                         "chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22","chrX","chrY","chrM")
+            myVchr <- unique(chr_)
+            if (! all(myVchr %in% typeChr)) {
+                # drop file
+                chrValid <- FALSE
+            } else {
+                chrValid <- TRUE
+            }
+
+            if (classValid & typeValid & chrValid == FALSE) {
```

Comment:
:-)

```
if (! classValid && typeValid && chrValid) ...
```
(... or "&" if you must, but that's confusing in an if-statement, it just happens to work here.)